numpyro/util.py

```diff
@@ -408,3 +408,121 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
         lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
     )
     return tree_map(lambda y: jnp.reshape(y, batch_shape + jnp.shape(y)[1:]), ys)
+
+
+def format_shapes(trace, *, title="Trace Shapes:", last_site=None, log_prob=False):
+    """
+    Returns a string showing a table of the shapes of all sites in the
+    trace.
+
+    :param dict trace: The model trace to format. Use ``numpyro.handlers.trace`` to
+        produce the trace.
+    :param str title: Title for the table of shapes.
+    :param str last_site: Name of a site in the model. If supplied, subsequent sites
+        are not displayed in the table.
+    :param bool log_prob: Display shapes of log probabilities for each sample site.
```

Comment:
I think we can rename `log_prob` to [compute_log_prob](https://docs.pyro.ai/en/stable/_modules/pyro/poutine/trace_struct.html#Trace.compute_log_prob) which supports True/False/callable where callable can be used to filter the sites.